File: classes/201-f25/Comparing-Python-and-Kotlin/conditionals.py
"""
   conditionals.py
   Jeff Ondich, 2013-01-03
   Anna Rafferty

   Modified by Tanya Amert for Spring 2025

   A tiny illustration of if-statements and console input.

   Intended as the Python half of parallel examples in Python and
   Kotlin. See Conditionals.kt.
"""

# Get the number from the user
numberString = input("Number, please: ")
number = int(numberString)

# Print a message based on the number
if number > 100:
    print(f"Gosh, {number} is a big number")
elif number == 3:
    print("3 is my lucky number")
elif number >= 0:
    print(f"{number} is pretty little")
else:
    print(f"{number} is negative")

File: classes/201-f25/Comparing-Python-and-Kotlin/Conditionals.kt
/*
 * Conditionals.kt
 * Tanya Amert, 2025-03-31
 * Based on Conditionals.java by Jeff Ondich, Anna Rafferty
 *
 * A tiny illustration of if-statements and console input.
 *
 * This is the Kotlin half of a pair of parallel examples in Python and Kotlin.
 * See conditionals.py.
 */

fun main() {

    // Get the number from the user
    print("Number, please: ")
    val numberString: String? = readLine()   // it could be null
    val number: Int = numberString!!.toInt() // so we assume it's not, forcefully :)

    // Print a message based on the number
    when {
        number > 100 -> println("Gosh, $number is a big number")
        number == 3 -> println("3 is my lucky number")
        number >= 0 -> println("$number is pretty little")
        else -> println("$number is negative")
    }
}